db/migrate/20240712094833_create_academic_years.rb

```diff
@@ -0,0 +1,25 @@
+class CreateAcademicYears < ActiveRecord::Migration[7.1]
+  class AcademicYearForMigration < ApplicationRecord
+    self.table_name = :academic_years
+  end
+
+  def change
+    create_table :academic_years, id: :uuid do |t|
+      t.string :name
+      t.date :starts_on
+      t.date :ends_on
+
+      t.timestamps
+    end
+
+    up_only do
+      [2023, 2024, 2025].each do |year|
+        AcademicYearForMigration.create!(
+          name: "#{year} to #{year + 1}",
+          starts_on: Date.new(year, 8, 1),
+          ends_on: Date.new(year + 1, 7, 31),
```

Comment:
Are these dates definitely correct?

- 1 August 2023 – 31 July 2024

Schools usually start around 3 September, not August. Is 1 August an 'official' date?

Reply:
I've used [this](https://www.gov.uk/guidance/understanding-academic-years) as a reference for these dates.

Reply:
Thanks for sharing that link. Although [that page says](https://www.gov.uk/guidance/understanding-academic-years#academic-year-start-and-end-dates):

> For the majority of students, each academic year will run from 1 September to 31 August but there are 4 official start dates in each academic year with each lasting 12 months.
> 
> If your course starts between: | Your academic year will run from:
> -- | --
> 1 August and 31 December | 1 September to 31 August
> 1 January and 31 March | 1 January to 31 December
> 1 April and 30 June | 1 April to 31 March
> 1 July and 31 July | 1 July to 30 June

Which is pretty unclear.

But it sounds like "the majority of students" it should be 1 September – 31 August?